reark/src/main/java/io/reark/reark/data/store/ContentProviderStore.java

```diff
@@ -57,66 +64,124 @@
  * @param <T> Type of the data this store contains.
  */
 public abstract class ContentProviderStore<T> {
-    private static final String TAG = ContentProviderStore.class.getSimpleName();
+    private final String TAG = getClass().getSimpleName();
 
     @NonNull
     private final ContentResolver contentResolver;
 
-    @NonNull
     private final ContentObserver contentObserver = getContentObserver();
-
-    @NonNull
     private final PublishSubject<Pair<T, Uri>> updateSubject = PublishSubject.create();
+    private final ObjectLockHandler<Uri> locker = new ObjectLockHandler<>();
+    private final int groupingTimeout;
+    private final int groupMaxSize;
 
     protected ContentProviderStore(@NonNull ContentResolver contentResolver) {
+        this(contentResolver, 100, 30);
+    }
+
+    protected ContentProviderStore(@NonNull ContentResolver contentResolver, int groupingTimeout, int groupMaxSize) {
         Preconditions.checkNotNull(contentResolver, "Content Resolver cannot be null.");
 
         this.contentResolver = contentResolver;
         this.contentResolver.registerContentObserver(getContentUri(), true, contentObserver);
 
-        updateSubject
-                .onBackpressureBuffer()
-                .observeOn(Schedulers.io())
-                .subscribe(pair -> {
-                    updateIfValueChanged(this, pair);
-                });
+        this.groupingTimeout = groupingTimeout;
+        this.groupMaxSize = groupMaxSize;
+
+        // Observable transforming updates and inserts to ContentProviderOperations
+        Observable<ContentProviderOperation> operationObservable = updateSubject
+                .flatMap(pair -> createOperation(pair.first, pair.second))
+                .filter(new NullFilter())
+                .observeOn(Schedulers.computation());
```

Comment:
I think the idea what that the things inside groupOperations() would be done in computation thread. Probably not needed anyway.